src/execution/__tests__/variables-test.ts

```diff
@@ -1434,5 +1461,33 @@ describe('Execute: Handles inputs', () => {
         },
       });
     });
+
+    it('when argument passed to a directive', () => {
+      const result = executeQueryWithFragmentArguments(`
+        query {
+          ...a(value: true)
+        }
+        fragment a($value: Boolean!) on TestType {
+          fieldWithNonNullableStringInput @skip(if: $value)
+        }
+      `);
+      expect(result).to.deep.equal({
+        data: {},
+      });
+    });
+
+    it('when a nullable argument to a directive with a field default is not provided and shadowed by an operation variable', () => {
```

Comment:
The point of the test is that we are providing a default for our custom skip directive and so that default should be grabbed just as it is in the tests above.

https://github.com/yaacovCR/graphql-executor/blob/1ea79ef6c0a29d521c51252106ac4af2ba55a7b0/src/execution/__tests__/variables-test.ts#L177-L178

In the light of day, however, it appears that `collectFields()` does not call `schema.getDirective('skip')` to find our custom skip directive. I am not 100% sure that is the correct behavior, but it seems besides the point right now.[^1]

In any case, if you manually change the reference in `collectFields()` to `schema.getDirective('skip')` instead of `GraphQLSkipDirective`, it should pick up our default, and it doesn't, see https://github.com/JoviDeCroock/graphql-js/pull/2#discussion_r1730094734

I will change the test, though, because it is kind of worthless in the end considering how collectFields doesn't pick up that default that we added!

[^1]: Generally speaking, It is valid to have custom runtime skip/include/defer/stream directives, although within the graphql-js explicitly any custom runtime directives would presumably have a custom version of `collectFields`, so it seems kind of harmless, except that it prevents us from adding a default value to `@skip(if:)` like in our test. Of course, `@skip(if:)` has a default value of `true` because when the argument is omitted, it will just skip by design, so it doesn't need a default. I only added the default to check how fragment variables will work. So I don't think it's really worth it to "fix" `collectFields()`. Calling  `schema.getDirective('skip')` is just one extra function call that isn't really necessary.